core/store/src/db/refcount.rs

```diff
@@ -112,27 +112,55 @@ pub(crate) fn encode_negative_refcount(rc: std::num::NonZeroU32) -> [u8; 8] {
 /// Assumes that all provided values with positive reference count have the same
```

Comment:
Question about the rules above - why do we even need to keep track of rc < 0? Could operands be applied in different order where we first delete a (key, value) and add it later?

Reply:
I think that's because originally this code was conflating the partial and full merges together. So a partial merge would run first to merge two adjacent operators, such as

* +1
* -2

to arrive at `-1` which only then is applied to the refcount stored in the storage.

Now that this function is only handling full merges, the negative intermediate refcounts should not exist at all, otherwise there's a bug somewhere that's emitting these refcount changes.